src/components/protocol_handler/src/protocol_packet.cc

```diff
@@ -319,34 +318,57 @@ RESULT_CODE ProtocolPacket::ProtocolHeaderValidator::validate(
   // For Control frames Data Size value shall be less than MTU header
   // For Single and Consecutive Data Size value shall be greater than 0x00
   // and shall be less than payload size
-  if (header.dataSize > payload_size) {
-    LOG4CXX_WARN(logger_,
-                 "Packet data size is "
-                     << header.dataSize
-                     << " and bigger than allowed payload size " << payload_size
-                     << " bytes");
-    return RESULT_FAIL;
-  }
+  // For First Frame Data Size shall be equal 0x08 (payload of this packet will
+  // be 8 bytes).
+
   switch (header.frameType) {
     case FRAME_TYPE_SINGLE:
     case FRAME_TYPE_CONSECUTIVE:
-      if (header.dataSize <= 0u) {
+      if (header.dataSize <= 0u || header.dataSize > payload_size) {
+        LOG4CXX_WARN(
+            logger_,
+            "Packet data size of "
+                << (header.frameType == FRAME_TYPE_SINGLE
+                        ? "FRAME_TYPE_SINGLE"
+                        : "FRAME_TYPE_CONSECUTIVE")
+                << " frame must be in range (0, payload_size=" << payload_size
```

Comment:
@KVGrygoriev I think you can write here just:
```
<< " frame must be in range (0, " << payload_size << ") ..."
```

Reply:
@AKalinich-Luxoft in range notation this means 0 < data_size <= payload_size --->>> (0, payload_size]

Reply:
@EKuliiev 
I guess he meant
"frame must be in range (0, payload_value]"
instead of
"frame must be in range (0, payload_size=payload_value]"

Reply:
@KVGrygoriev ok. more descriptive, less descriptive log

Reply:
the same conclusion

Reply:
@EKuliiev I agree with range notation, there are no mistakes. @KVGrygoriev understand what I meant